### ISSUE TITLE
GHC HEAD warns about RULE possibly not firing

### ISSUE THREAD

**Issue by hvr** [kolmodin/binary#98]
```
src/Data/Binary/Get/Internal.hs:395:1: warning:
    Rule "<$> to <*>" may never fire because ‘<$>’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma for ‘<$>’
```

**Comment by hvr** [comment]
curiously, I don't see the warnings mentioned in #82

**Comment by kolmodin** [comment]
I've already worked some at removing warnings :)

**Comment by kolmodin** [comment]
Fixed with 11816f1fb2400171d018b784b9522604819b8778.
